socketio_middleware/src/rooms.rs

```diff
@@ -33,35 +35,114 @@ impl ChannelPair {
 }
 
 pub fn join_channel_to_room(room_id: &str, channel_pair: ChannelPair) {
+    //TODO: why use remove, it is low performance.
     let mut connected_sockets = match ROOMS.remove(room_id) {
         Some(val) => val,
         None => Vec::new(),
     };
 
-    connected_sockets.push(channel_pair);
+    //check if socketid exist
+    let mut exist: bool = false;
+    for socket in &connected_sockets {
+        if socket.sid() == channel_pair.sid() {
+            debug!("ROOMS: socketid {} doesn't join into room {}, this socketid already exist in the room.", channel_pair.sid(), room_id);
+            exist = true;
+            break;
+        }
+    }
 
+    if false == exist {
+        debug!("ROOMS: socketid {} joined into room {}.", channel_pair.sid(), room_id);
+        connected_sockets.push(channel_pair);
+    }
     ROOMS.insert(room_id.to_string(), connected_sockets);
+
+    //print all sockets in the room
+    match ROOMS.get(room_id) {
```

Comment:
This all looks like debug code, but could slow down the code when we're not in debug mode.

Reply:
yes, they are only debug info. I have disable them.

Reply:
I create a new function "print_sockets_in_room" , app can call this function for debug information.

Removed from join/leave room.